bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/sdm/api/PubSubAPI.java

```diff
@@ -153,14 +155,21 @@ public PubSubAPI(HttpClientFactory httpClientFactory, OAuthFactory oAuthFactory,
             String clientId, String clientSecret) {
         this.httpClient = httpClientFactory.getCommonHttpClient();
         this.projectId = projectId;
-        this.oAuthService = oAuthFactory.createOAuthClientService(String.format(PUBSUB_HANDLE_FORMAT, ownerId),
-                TOKEN_URL, AUTH_URL, clientId, clientSecret, PUBSUB_SCOPE, false);
+        this.oAuthFactory = oAuthFactory;
+        this.oAuthServiceHandleId = String.format(PUBSUB_HANDLE_FORMAT, ownerId);
+        this.oAuthService = oAuthFactory.createOAuthClientService(oAuthServiceHandleId, TOKEN_URL, AUTH_URL, clientId,
+                clientSecret, PUBSUB_SCOPE, false);
         scheduler = new ScheduledThreadPoolExecutor(3, new NamedThreadFactory(ownerId, true));
     }
 
     public void dispose() {
         subscriptionListeners.clear();
         scheduler.shutdownNow();
+        oAuthFactory.ungetOAuthService(oAuthServiceHandleId);
+    }
+
+    public void deleteServiceAndAccessToken() {
```

Comment:
Perhaps to add some context:
```suggestion
    public void deleteOAuthServiceAndAccessToken() {
```

Reply:
Method renamed